Add rowSx (#8986)
marmelab/react-admin: docs/SimpleList.md

);
```

## `rowSx`

> This optional prop should be a function, which gets called for each row. It receives the current record and index as arguments, and should return a style object. The style object is applied to the `<ListItem>` styles prop.

Reviewer: ```suggestion
This optional prop should be a function, which gets called for each row. It receives the current record and index as arguments, and should return an object accepted by Material UI [`sx` prop](https://mui.com/system/getting-started/the-sx-prop/). The object is applied to the `<ListItem>` `sx` prop.
```
